src/Domain/FunctionsOfTime/PiecewisePolynomial.cpp

```diff
@@ -206,13 +208,14 @@ void PiecewisePolynomial<MaxDeriv>::update(
   // to ensure everything is consistent. This gives us the ordering above.
   deriv_info_at_update_times_.emplace_back(time_of_update, std::move(func));
   deriv_info_size_.fetch_add(1, std::memory_order_acq_rel);
-  expiration_time_.store(next_expiration_time, std::memory_order_release);
+  expiration_time_.compare_exchange_strong(
```

Comment:
Yep, then check that the result is `true`:
```cpp
if (not expiration_time_.compare_exchange_strong(
    current_expiration_time, next_expiration_time, std::memory_order_acq_rel)) {
  ERROR("...");
}
```

Reply:
Squashed into the current fixup for @wthrowe to view.